libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -98,14 +99,25 @@ internal class RouteAlternativesController constructor(
             // TODO make async
             val alternatives: List<DirectionsRoute> = runBlocking {
                 routeAlternatives.mapIndexedNotNull { index, routeAlternative ->
-                    val alternative = parseNativeDirectionsAlternative(
-                        routeAlternative.routeResponse,
-                        routeProgress.route.routeOptions()
-                    )
-                    if (alternative == null) {
-                        logE(TAG, Message("null alternative at index $index"))
+                    try {
+                        val alternative = parseNativeDirectionsAlternative(
+                            ThreadController.IODispatcher,
+                            routeAlternative.routeResponse,
+                            routeProgress.route.routeOptions()
+                        )
+                        alternative
+                    } catch (ex: Exception) {
```

Comment:
Catching arbitrary `Exception` is usually not a good idea. 
E.g. catching `CancellationException` can lead to problems with cancellation of coroutines.